lib/build-task.ts

```diff
@@ -83,4 +83,8 @@ export interface BuildTask {
 	 * the composition
 	 */
 	serviceName: string;
+	/**
+	 * Has this task failed to be resolved?
+	 */
+	resolveFailure: boolean;
```

Comment:
I think `resolveFailure` makes me expect it's of `Error` type, not a boolean -- let's either store the `ProjectResolutionError` here (if you think that contains useful info) or rename the attribute to `resolved`.

Reply:
Yeah it's a bit weird with `BuildTask` being an interface, this need to be defined first, but you're right having it intialised with `false` makes more sense and naming it `resolved`.